sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -942,6 +957,47 @@ class Analyzer(
       }
   }
 
+  /**
+   * Replace [[UnresolvedRelation]] with concrete streaming logical plans.
+   */
+  object ResolveStreamingRelation extends Rule[LogicalPlan] {
+    override def apply(plan: LogicalPlan): LogicalPlan = plan.resolveOperatorsUp {
+      case u: UnresolvedRelation if u.isStreaming =>
+        val res = lookupStreamingRelation(u.multipartIdentifier, u.options)
+        res.getOrElse(u)
+    }
+
+    // Look up a relation from the session catalog with the following logic:
+    // 1) If the resolved catalog is not session catalog, return None.
+    // 2) If a relation is not found in the catalog, return None.
+    // 3) If a v1 table is found, create a v1 relation. Otherwise, pass the table to
+    //    UnresolvedStreamingRelation.
```

Comment:
hmm, I don't see `UnresolvedStreamingRelation`, is it missing?

Reply:
Old comments for prototype, the whole rule has been merged with ResolveRelation in 305c316.